oauth2client/_helpers.py

```diff
@@ -300,3 +295,55 @@ def _urlsafe_b64decode(b64string):
     b64string = _to_bytes(b64string)
     padded = b64string + b'=' * (4 - len(b64string) % 4)
     return base64.urlsafe_b64decode(padded)
+
+
+def update_query_params(uri, params):
+    """Updates a URI with new query parameters.
+
+    If a given key from ``params`` is repeated in the ``uri``, then
```

Comment:
I don't have an opinion on what should happen, so just tell me what you want. The issue is that casting the results of `parse_qsl` (a list of tuples, with repeats allowed) directly to a `dict` is that query params can be lost.

It's possible / probable that all call sites assumed the same thing that `parse_unique_urlencoded` does and then meant to replace (if existing) or add the param.